File: examples/sync_support/okx/okx_public.py
import dcex

client = dcex.okx()

result = client.get_public_instruments(instType="SPOT")
print(result)

result = client.get_funding_rate(product_symbol="BTC-USDT-SWAP")
print(result)

result = client.get_funding_rate_history(product_symbol="BTC-USDT-SWAP")
print(result)

File: crates/dcex/examples/okx_public.rs
use std::time::Duration;

use dcex::exchanges::okx::OkxClient;

#[tokio::main]
async fn main() -> dcex::Result<()> {
    let client = OkxClient::public(Duration::from_secs(10))?;
    let response = client.get_orderbook("BTC-USDT-SPOT").sz(5).await?;
    println!("{}", response.data);
    Ok(())
}
